### PR TITLE
Add type mapping documentation

### LINKED ISSUE #2370: Add documentation for reference types
Then let's add a new paragraph into `schema-types.md` like 
```
##References
```

and describe these 3 classes. These are special "service" types that are worth mentioning among all the others types.

_Originally posted by @sungam3r in https://github.com/graphql-dotnet/graphql-dotnet/issues/2367#issuecomment-792767974_

### PR DESCRIPTION
**Pull request by Shane32**
Fixes #2370

### CODE REVIEW

File: docs2/site/docs/getting-started/schema-types.md [hunk 1]
```diff
@@ -276,3 +276,66 @@ be implemented as:
...
+public class MySchema
+{
+    public void MySchema()
+    {
+        ...
+
+        // For output graphs, map the 'User' data model class to the output object graph type 'UserGraphType'
+        this.RegisterTypeMapping<User, UserGraphType>();
+
+        // For input graphs, map the 'User' data model class to the input object graph type 'UserGraphType'
```
Comment (sungam3r) [inline docs2/site/docs/getting-started/schema-types.md:307]:
```suggestion
        // For input graphs, map the 'User' data model class to the input object graph type 'UserInputGraphType'
```

File: docs2/site/docs/getting-started/schema-types.md [hunk 2]
```diff
@@ -276,3 +276,66 @@ be implemented as:
         }
     }
 ```
+
+## Type Mapping
```
Comment (sungam3r) [inline docs2/site/docs/getting-started/schema-types.md:280]: See `Ability to map CLR types to GraphTypes` section in v4 migration guide. It's worth to mention `GraphQLMetadata` attribute here as an alternative way to define the mapping.

Comment (Shane32) [reply docs2/site/docs/getting-started/schema-types.md:280]: You're right, it should be mentioned here.  Ugh I thought I was done.  Well, maybe I can get to it tomorrow night.

Comment (Shane32) [reply docs2/site/docs/getting-started/schema-types.md:280]: Done

File: docs2/site/docs/getting-started/schema-types.md [hunk 3]
```diff
@@ -276,3 +276,76 @@ be implemented as:
...
+
+When specifying a field using the shortcut syntax `Field(x => x.Parent)`, which does not specify
+a specific graph type, GraphQL.NET will first look at the data model to see if it has a `GraphQLMetadata`
+attribute specified on it indicating the graph type to use for the data model. For instance, you can
+specify the graph type for a `Widget` class in the following manner:
+
+```csharp
+[GraphQLMetadata(InputType = typeof(WidgetInputGraphType), OutputType = typeof(WidgetOutputGraphType)]
```
Comment (Shane32) [inline docs2/site/docs/getting-started/schema-types.md:288]:
```suggestion
[GraphQLMetadata(InputType = typeof(WidgetInputGraphType), OutputType = typeof(WidgetGraphType)]
```

**Review by sungam3r — APPROVED** [review]

**Pull request merged**